src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                double firstLeg = random.nextDouble() * 10 + 1;
+                double secondLeg = random.nextDouble() * 10 + 1;
+                double trapezoidHeight = random.nextDouble() * 10 + 1;
+                return new IsoscelesTrapezoid(color, firstLeg, secondLeg, trapezoidHeight);
+            case 1:
+                double sideLength = random.nextDouble() * 10 + 1;
+                return new Square(color, sideLength);
```

Comment:
You've correctly created a variable for the side length before passing it to the constructor, which improves code readability.